File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}
        for s in strs:
            x = ''.join(sorted(s))
            if (x in d):
                d[x].append(s)
            else:
                d[x] = [s]
        return d.values()
        

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for (String word: strs){
            char[] ar = word.toCharArray();
            Arrays.sort(ar);
            String wordSorted = new String(ar);
            if (!map.containsKey(wordSorted)){
                map.put(wordSorted, new ArrayList<>());
            }
            map.get(wordSorted).add(word);
        }
        return new ArrayList<>(map.values());
    }
}